File: eRents.Shared/DTO/Response/ImageResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eRents.Shared.DTO.Response
{
	public class ImageResponse
	{
		public int ImageId { get; set; }
		public string FileName { get; set; }
		public DateTime DateUploaded { get; set; }
		public string Url { get; set; }  // URL to access the image via API endpoint
		public byte[]? ImageData { get; set; }  // Optional: only include when specifically requested
		public string? ContentType { get; set; }  // MIME type (e.g., "image/jpeg")
		public int? Width { get; set; }
		public int? Height { get; set; }
		public long? FileSizeBytes { get; set; }
		public bool IsCover { get; set; }  // Is this the cover/primary image
	}
}

File: eRents.UI/e_rents_mobile/lib/core/models/image_response.dart
import 'dart:typed_data';

class ImageResponse {
  final int imageId;
  final String fileName;
  final ByteData imageData ;
  final DateTime dateUploaded;

  ImageResponse({
    required this.imageId,
    required this.fileName,
    required this.imageData,
    required this.dateUploaded,
  });

  factory ImageResponse.fromJson(Map<String, dynamic> json) {
    return ImageResponse(
      imageId: json['imageId'],
      fileName: json['fileName'],
      imageData: json['imageData'],
      dateUploaded: DateTime.parse(json['dateUploaded']),
    );
  }
}